### PR TITLE
Add network status monitoring

### PR DESCRIPTION
**Pull request by Almond-Peach**

### CODE REVIEW

File: app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt [hunk 1]
```diff
@@ -3,14 +3,52 @@ package com.simply.birthdayapp.presentation
...
+                    initialValue = networkMonitor.isCurrentlyConnected(),
+                    lifecycleOwner = this,
+                )
+                var keepShowingNetworkDialog by rememberSaveable { mutableStateOf(true) }
+                var showNetworkDialog by rememberSaveable { mutableStateOf(true) }
+                val networkDialogRepeatDelayMillis = 10_000L
+
+                LaunchedEffect(showNetworkDialog) {
+                    if (keepShowingNetworkDialog && showNetworkDialog.not()) {
+                        lifecycleScope.launch {
```
Comment (artur-ghonyan) [inline app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt]: Why do you need to open 'scope' here?

Comment (Almond-Peach) [reply app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt]: Removed.

File: app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt [hunk 2]
```diff
@@ -3,14 +3,48 @@ package com.simply.birthdayapp.presentation
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             BirthdayAppTheme {
+                val connectedToNetwork by networkMonitor.connected.collectAsStateWithLifecycle(
+                    initialValue = networkMonitor.isCurrentlyConnected(),
+                    lifecycleOwner = this,
+                )
+                var keepShowingNetworkDialog by rememberSaveable { mutableStateOf(true) }
+                var showNetworkDialog by rememberSaveable { mutableStateOf(true) }
```
Comment (artur-ghonyan) [inline app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt]: Why are the default values set to 'true'?

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/components/NoNetworkConnectionDialog.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import com.simply.birthdayapp.presentation.ui.theme.AppTheme
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun NoNetworkConnectionDialog(
+    onDoNotShow: () -> Unit = {},
+    onDismiss: () -> Unit = {},
+) {
+    AlertDialog(onDismissRequest = onDismiss) {
+        Surface(
```
Comment (artur-ghonyan) [inline app/src/main/java/com/simply/birthdayapp/presentation/ui/components/NoNetworkConnectionDialog.kt]: Do you need 'Surface' here?

File: app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt (hunk 2 above)
Comment (Almond-Peach) [reply app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt]: By default, a dialog appears when the user is disconnected.

The dialog can be `dismissed` and `shown again` after 10 seconds.
The dialog `will not be shown` again until the app is restarted if the user clicks `Don't show`.

That is why these default values are set to `true`.

P.S. Have renamed that variables to align them with the user actions `Do not show`, `Dismiss`.

File: app/src/main/java/com/simply/birthdayapp/presentation/ui/components/NoNetworkConnectionDialog.kt (hunk 3 above)
Comment (Almond-Peach) [reply app/src/main/java/com/simply/birthdayapp/presentation/ui/components/NoNetworkConnectionDialog.kt]: Removed.

**Review by artur-ghonyan — APPROVED** [review]

**Pull request merged**